src/swarm-topography/Swarm/Game/Scenario/Topography/Area.hs

```diff
@@ -94,8 +112,13 @@ getAreaDimensions cellGrid =
 computeArea :: AreaDimensions -> Int32
 computeArea (AreaDimensions w h) = w * h
 
+-- |
+-- Warning: size must be nonzero,
```

Comment:
This function seems unnecessarily partial.  Why not have a special case to first check whether `w` or `h` is zero, and if so return `EmptyGrid`?